/*
 * Copyright 2013 Google Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Author: morlovich@google.com (Maksim Orlovich)

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package net_instaweb;

// This represents aggregated list of all selectors of the CSS that was needed
// to render the above-the-fold portion of a page.
//
// Next free id: 2
// TODO(morlovich): We may want to keep track of information on inputs to detect
// case of HTML inputs. We may also need history information, to deal with
// noisiness of the data.
message CriticalSelectorSet {
  // The set of all selectors that were guarding the CSS that got applied for
  // the above-the-fold portion of the webpage.
  //
  // Duplicates should not be included. In case of something using a
  // pseudo-element (e.g. foo::before), only the base portion should be
  // included.
  repeated string critical_selectors = 1;
}
